grpc/tests/src/test/deploy/receipts.rs

```diff
@@ -0,0 +1,440 @@
+
+use casper_engine_test_support::{
+    internal::{ExecuteRequestBuilder, InMemoryWasmTestBuilder, DEFAULT_RUN_GENESIS_REQUEST},
+    DEFAULT_ACCOUNT_ADDR,
+};
+use casper_types::{
+    account::AccountHash, runtime_args, AccessRights, DeployHash, PublicKey, RuntimeArgs, Transfer,
+    TransferAddr, U512,
+};
+
+const CONTRACT_TRANSFER_PURSE_TO_ACCOUNT: &str = "transfer_purse_to_account.wasm";
+const TRANSFER_ARG_TARGET: &str = "target";
+const TRANSFER_ARG_AMOUNT: &str = "amount";
+
+const CONTRACT_TRANSFER_PURSE_TO_ACCOUNTS: &str = "transfer_purse_to_accounts.wasm";
+const TRANSFER_ARG_SOURCE: &str = "source";
+const TRANSFER_ARG_TARGETS: &str = "targets";
+
+const CONTRACT_TRANSFER_PURSE_TO_ACCOUNTS_STORED: &str = "transfer_purse_to_accounts_stored.wasm";
+const CONTRACT_TRANSFER_PURSE_TO_ACCOUNTS_SUBCALL: &str = "transfer_purse_to_accounts_subcall.wasm";
+
+const ALICE_KEY: PublicKey = PublicKey::Ed25519([3; 32]);
+const BOB_KEY: PublicKey = PublicKey::Ed25519([5; 32]);
+const CAROL_KEY: PublicKey = PublicKey::Ed25519([7; 32]);
+
+lazy_static! {
+    static ref ALICE_ADDR: AccountHash = ALICE_KEY.into();
+    static ref BOB_ADDR: AccountHash = BOB_KEY.into();
+    static ref CAROL_ADDR: AccountHash = CAROL_KEY.into();
+    static ref TRANSFER_AMOUNT_1: U512 = U512::from(100_000_000);
+    static ref TRANSFER_AMOUNT_2: U512 = U512::from(200_000_000);
+    static ref TRANSFER_AMOUNT_3: U512 = U512::from(300_000_000);
+}
+
+#[ignore]
+#[test]
+fn should_record_wasmless_transfer() {
+    let mut builder = InMemoryWasmTestBuilder::default();
+    builder.run_genesis(&DEFAULT_RUN_GENESIS_REQUEST);
+
+    let transfer_request = ExecuteRequestBuilder::transfer(
+        *DEFAULT_ACCOUNT_ADDR,
+        runtime_args! {
+            TRANSFER_ARG_TARGET => *ALICE_ADDR,
+            TRANSFER_ARG_AMOUNT => *TRANSFER_AMOUNT_1
+
+        },
+    )
+    .build();
+
+    let deploy_hash = {
+        let deploy_items: Vec<DeployHash> = transfer_request
+            .deploys()
+            .iter()
+            .map(Result::as_ref)
+            .filter_map(Result::ok)
+            .map(|deploy_item| deploy_item.deploy_hash)
+            .collect();
+        deploy_items[0]
+    };
+
+    builder.exec(transfer_request).commit().expect_success();
+
+    let default_account = builder
+        .get_account(*DEFAULT_ACCOUNT_ADDR)
+        .expect("should have default account");
+
+    let alice_account = builder
+        .get_account(*ALICE_ADDR)
+        .expect("should have Alice's account");
+
+    let alice_attenuated_main_purse = alice_account
+        .main_purse()
+        .with_access_rights(AccessRights::ADD);
+
+    let deploy_info = builder
+        .get_deploy_info(deploy_hash)
+        .expect("should have deploy info");
+
+    assert_eq!(deploy_info.deploy_hash, deploy_hash);
+    assert_eq!(deploy_info.from, *DEFAULT_ACCOUNT_ADDR);
+    assert_eq!(deploy_info.source, default_account.main_purse());
+
+    // TODO: this is borked
```

Comment:
How do you mean?  Don't we want the gas cost to be zero if we're not using system contracts?

Reply:
My understanding was that Wasm-less transfer was supposed to be free, but it looks like if we run using system contracts, it is not.  I will file a bug.

Reply:
Ah - I see.  Thanks.